deploy/eck-operator/templates/statefulset.yaml

```diff
@@ -174,6 +174,12 @@ spec:
       {{- if .Values.hostNetwork }}
       hostNetwork: true
       dnsPolicy: ClusterFirstWithHostNet
+      {{- else if and (.Values.dnsPolicy) (.Values.dnsConfig) }}
+      dnsPolicy: {{ .Values.dnsPolicy }}
+      {{- end }}
+      {{- if and (.Values.dnsConfig) (.Values.dnsPolicy) }}
+      dnsConfig:
+        {{- .Values.dnsConfig | toYaml | nindent 8 }}
```

Comment:
I'm not sure I like explicitly linking `dnsPolicy` and `dnsConfig` as we are doing here. Why not just set these fields when they are set in the values file and allow the user full control over these settings?

```
{{- with .Values.dnsPolicy }}
dnsPolicy: {{ .Values.dnsPolicy }} 
{{- end }}
{{- with .Values.dnsConfig }}
dnsConfig:
{{- toYaml . | nindent 8 }}
{{- end }}
```